types/wlr_output_layout.c

```diff
@@ -421,3 +421,53 @@ struct wlr_output *wlr_output_layout_get_center_output(
 
 	return wlr_output_layout_output_at(layout, dest_x, dest_y);
 }
+
+
+struct wlr_output *wlr_output_layout_adjacent_output(
+		struct wlr_output_layout *layout, enum wlr_direction direction,
+		struct wlr_output *reference, double ref_x, double ref_y) {
+	// XXX should we allow reference to be NULL?
```

Comment:
Why?

Reply:
The only reason I can think of is to make it consistent with the rest of the api. The rule generally is that if reference is not passed, then it acts on the whole layout, but there's no concrete use case for this yet.

Reply:
I don't know how that could work, probably best not to.

Reply:
Should drop the comment, leave the assert

Reply:
Can't we automatically get the `reference` output from `ref_x` and `ref_y`?

Reply:
We could, but we need to start being more careful about supporting the overlapping output case. There are a few functions that currently don't support this well that need to be revisited.